dotnet/src/Connectors/Connectors.AI.OpenAI/OpenAIRequestSettings.cs

```diff
@@ -63,7 +63,22 @@ public class OpenAIRequestSettings : AIRequestSettings
     /// Use carefully and ensure that you have reasonable settings for max_tokens and stop.
     /// </summary>
     [JsonPropertyName("results_per_prompt")]
-    public int ResultsPerPrompt { get; set; } = 1;
+    public int ResultsPerPrompt
+    {
+        get
+        {
+            if (this.ExtensionData.TryGetValue("results_per_prompt", out var value))
```

Comment:
If this property is applicable to general SemanticFunction logic (Method `GetResultsCountOrDefault`), we can:
- Move `ResultsPerPrompt` parameter to base `AIRequestSettings` with default value `1` - although it will mean that this parameter is applicable to every AI connector, which is not the case.
- Place logic for multiple results only where it's applicable - move it from SemanticFunction to OpenAI Connector.
- Remove this property from `OpenAIRequestSettings` completely and place it as parameter of function. (I had an idea to replace SKContext as input parameter for function/kernel with `FunctionRequest`/`KernelRequest`, and property `ResultsPerPrompt` can be good first candidate for request models, but this is definitely separate scope of work and PR.)

Main point here, that we should not use `ExtensionData` as a way of storing general settings for functions.